tests/Graphql/Resolver/CollectionResolverFactoryTest.php

```diff
@@ -123,66 +161,55 @@ public function testCreatePaginatorCollectionResolver()
         $collectionPaginatorProphecy->count()->willReturn(8);
         $collectionPaginatorProphecy->getItemsPerPage()->willReturn(8);
 
-        $mockedCollectionResolverFactory = $this->mockCollectionResolverFactory($collectionPaginatorProphecy->reveal(), [], [], true);
+        $resolverFactory = $this->createCollectionResolverFactory($collectionPaginatorProphecy->reveal(), [], [], true);
+        $resolver = $resolverFactory(RelatedDummy::class, Dummy::class, 'operationName');
 
-        $resolver = $mockedCollectionResolverFactory->createCollectionResolver('resourceClass', 'rootClass');
-
-        $resolveInfoProphecy = $this->prophesize(ResolveInfo::class);
-        $resolveInfoProphecy->fieldName = 'rootProperty';
+        $resolveInfo = new ResolveInfo([]);
+        $resolveInfo->fieldName = 'relatedDummies';
+        $resolveInfo->fieldNodes = [];
 
         $this->assertEquals(
             ['edges' => [['node' => 'normalizedObject1', 'cursor' => 'Mg==']], 'pageInfo' => ['endCursor' => 'MTY=', 'hasNextPage' => true]],
-            $resolver(null, ['after' => 'MQ=='], null, $resolveInfoProphecy->reveal())
+            $resolver(null, ['after' => 'MQ=='], null, $resolveInfo)
         );
     }
 
-    public function paginationProvider()
-    {
-        return [[true, ['edges' => [], 'pageInfo' => ['endCursor' => null, 'hasNextPage' => false]]], [false, []]];
-    }
-
-    public function cursorProvider()
-    {
-        return [['$bad$', ['MA==', 'MQ==']], ['MQ==', ['Mg==', 'Mw==']]];
-    }
-
-    public function subresourceProvider()
-    {
-        return [[['Subobject1', 'Subobject2'], ['normalizedSubobject1', 'normalizedSubobject2']], [[], []]];
-    }
-
-    private function mockCollectionResolverFactory($collection, array $subcollection, array $identifiers, bool $paginationEnabled): CollectionResolverFactory
+    /**
+     * @param array|\Iterator $collection
+     */
+    private function createCollectionResolverFactory($collection, array $subcollection, array $identifiers, bool $paginationEnabled): CollectionResolverFactory
     {
         $collectionDataProviderProphecy = $this->prophesize(CollectionDataProviderInterface::class);
-        $collectionDataProviderProphecy->getCollection('resourceClass')->willReturn($collection);
+        $collectionDataProviderProphecy->getCollection(Dummy::class)->willReturn($collection);
+        $collectionDataProviderProphecy->getCollection(RelatedDummy::class)->willReturn($collection);
 
         $subresourceDataProviderProphecy = $this->prophesize(SubresourceDataProviderInterface::class);
-        $subresourceDataProviderProphecy->getSubresource('subresourceClass', $identifiers, [
-            'property' => 'rootProperty',
-            'identifiers' => [['rootIdentifier', 'rootClass']],
+        $subresourceDataProviderProphecy->getSubresource(RelatedDummy::class, $identifiers, [
+            'property' => 'relatedDummies',
+            'identifiers' => [['id', Dummy::class]],
```

Comment:
Sometimes `id` isn't called `id` but is another property. Maybe we should add such test to prevent failing cases.

Reply:
`id` is mandatory according to the Relay spec.